src/material/datepicker/month-view.ts

```diff
@@ -252,6 +252,28 @@ export class MatMonthView<D> implements AfterContentInit, OnChanges, OnDestroy {
     this._changeDetectorRef.markForCheck();
   }
 
+  _dateBecomesActive(event: MatCalendarUserEvent<number>) {
```

Comment:
This is 90% identical to the ` _dateSelected` method above. This code could be more DRY. I'm open to suggestions on how to refactor it.

Here's what I had in mind. Create 2 reusable functions with shared code between `dateBecomesActive` and `_dateSelected` to reduce duplicated code.

```
// final variable names subject to bike shedding
_normalizeNumericDateToThisMonth(date: number, normalizeTo: D) {
     const year = this._dateAdapter.getYear(normalizeTo);
    const month = this._dateAdapter.getMonth(normalizeTo);
    return this._dateAdapter.createDate(year, month, date);
}

_didUserEventChangeActiveOrSelectedDate(event: MatCalendarUserEvent<number>, referenceDate: DateRange<D> | D | null) {
    let rangeStartDate: number | null;
    let rangeEndDate: number | null;
    
    if (referenceDate instanceof DateRange) {
      rangeStartDate = this._getDateInCurrentMonth(referenceDate.start);
      rangeEndDate = this._getDateInCurrentMonth(referenceDate.end);
    } else {
      rangeStartDate = rangeEndDate = this._getDateInCurrentMonth(referenceDate);
    }
    
    return rangeStartDate !== date || rangeEndDate !== date;
}
```

Reply:
I don't fully follow why these two things need to work the same way; in particular, the active date is always going to be a single date (whereas selection can be a range of dates). Shouldn't this be as direct as when a date-cell is focused, set that cell's date as the active date?

Reply:
that's right, the active date can only be a single date. I was confused. That means I can delete the daterange code from `_dateBecomesActive`, which will clean the code up a bit.

Reply:
I removed the unnecessary code.